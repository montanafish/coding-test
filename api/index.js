export const getShowTitles = (numOfRecords = 1) => {
  const showTitles = []

  for (let i = 0; i < numOfRecords; i += 1) {
    showTitles.push({
      id: i,
      title: 'Home and Away',
      description: 'THE ENDLESS SUMMER',
      image: {
        medium: '/static/images/tile-medium.jpg',
        large: '/static/images/tile.jpg'
      },
      icon: '/static/images/logo-40x40.png',
      seasons: 8,
      views: 5752532,
    })
  }
  return showTitles
}
