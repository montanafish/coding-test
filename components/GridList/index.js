import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import media from 'styles/media';
import { sizes } from 'styles/variables';

const Grid = styled.div`
  display: flex;
  flex-direction: ${props => (props.vertical ? 'column' : 'row')};
  justify-content: flex-start;
  flex-wrap: wrap;
  webkitoverflowscrolling: 'touch';

  ${media.tablet`
    justify-content: center;
  `} 
  
  > div {
    flex: 1 1 auto;
    margin: ${props => `${props.spacing}px`};
    max-width: ${props => `${props.itemWidth}px`};
    
    ${media.mobile`
      max-width: ${sizes.width.mobile}px;
    `};
  }
`;

class GridList extends React.PureComponent {
  render() {
    const { children, vertical, spacing, itemWidth } = this.props;

    return (
      <Grid vertical={vertical} spacing={spacing} itemWidth={itemWidth}>
        {React.Children.map(children, (child, i) => (
          <div key={i}>{child}</div>
        ))}
      </Grid>
    );
  }
}

GridList.propTypes = {
  children: PropTypes.node.isRequired,
  vertical: PropTypes.bool,
  spacing: PropTypes.number,
  itemWidth: PropTypes.number,
};

GridList.defaultProps = {
  spacing: 5,
  itemWidth: 315,
};

export default GridList;
