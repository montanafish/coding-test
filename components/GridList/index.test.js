/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { getShowTitles } from 'api';
import GridList from './index';

const showTitles = getShowTitles(3);

describe('<GridList />', () => {
  it('should render a div', () => {
    const wrapper = shallow(
      <GridList>
        <br />
      </GridList>
    );
    expect(wrapper.childAt(0).type()).toBe('div');
  });

  it('renders children', () => {
    const wrapper = shallow(
      <GridList>
        {showTitles.map(showTitle => (
          <span key={showTitle.id}>{showTitle.title}</span>
        ))}
      </GridList>
    );

    expect(wrapper.find('div')).toHaveLength(showTitles.length);
  });

  it('renders children and change spacing', () => {
    const spacing = 10;
    const component = renderer.create(
      <GridList spacing={spacing}>
        {showTitles.map(showTitle => (
          <span key={showTitle.id}>{showTitle.title}</span>
        ))}
      </GridList>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders children and change item width', () => {
    const itemWidth = 400;
    const component = renderer.create(
      <GridList itemWidth={itemWidth}>
        {showTitles.map(showTitle => (
          <span key={showTitle.id}>{showTitle.title}</span>
        ))}
      </GridList>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders children and change vertical', () => {
    const vertical = true;
    const component = renderer.create(
      <GridList vertical={vertical}>
        {showTitles.map(showTitle => (
          <span key={showTitle.id}>{showTitle.title}</span>
        ))}
      </GridList>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
