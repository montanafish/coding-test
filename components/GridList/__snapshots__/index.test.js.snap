// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridList /> renders children and change item width 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  webkitoverflowscrolling: 'touch';
}

.c0 > div {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  margin: 5px;
  max-width: 400px;
}

@media (max-width:768px) {
  .c0 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (max-width:767px) {
  .c0 > div {
    max-width: 414px;
  }
}

<div
  className="c0"
  spacing={5}
>
  <div>
    <span>
      Home and Away
    </span>
  </div>
  <div>
    <span>
      Home and Away
    </span>
  </div>
  <div>
    <span>
      Home and Away
    </span>
  </div>
</div>
`;

exports[`<GridList /> renders children and change spacing 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  webkitoverflowscrolling: 'touch';
}

.c0 > div {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  margin: 10px;
  max-width: 315px;
}

@media (max-width:768px) {
  .c0 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (max-width:767px) {
  .c0 > div {
    max-width: 414px;
  }
}

<div
  className="c0"
  spacing={10}
>
  <div>
    <span>
      Home and Away
    </span>
  </div>
  <div>
    <span>
      Home and Away
    </span>
  </div>
  <div>
    <span>
      Home and Away
    </span>
  </div>
</div>
`;

exports[`<GridList /> renders children and change vertical 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  webkitoverflowscrolling: 'touch';
}

.c0 > div {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  margin: 5px;
  max-width: 315px;
}

@media (max-width:768px) {
  .c0 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (max-width:767px) {
  .c0 > div {
    max-width: 414px;
  }
}

<div
  className="c0"
  spacing={5}
>
  <div>
    <span>
      Home and Away
    </span>
  </div>
  <div>
    <span>
      Home and Away
    </span>
  </div>
  <div>
    <span>
      Home and Away
    </span>
  </div>
</div>
`;
